app/controllers/application_templates_controller.rb

```diff
@@ -27,7 +27,24 @@ def toggle_active
   private
 
   def find_template
-    params.require :id
+    if params[:specific_path]
```

Comment:
Right, so this should be done via routing, not with controller method changes.

http://guides.rubyonrails.org/routing.html#dynamic-segments